templates/commands/cel.go

```diff
@@ -151,6 +155,55 @@ var celFuncs = []cel.EnvOption{
 			}),
 		),
 	),
+
+	// gcp_matches_project_number returns whether the input matches the format
+	// of a GCP project number (only digits, and not a tiny number).
+	//
+	// Examples:
+	//   gcp_matches_project_number("123456789")==true
+	//   gcp_matches_project_number("123abc")==false
+	cel.Function(
+		"gcp_matches_project_number",
+
+		// There are multiple implementations; this one accepts string.
+		cel.Overload("gcp_matches_project_number_string",
+			[]*types.Type{types.StringType},
+			cel.BoolType,
+			cel.UnaryBinding(func(input ref.Val) ref.Val {
+				strAny, err := input.ConvertToNative(reflect.TypeOf(""))
+				if err != nil {
+					return types.NewErr("internal error: argument was not convertible to string: %w", err)
+				}
+				str, ok := strAny.(string)
+				if !ok {
+					return types.NewErr("internal error: argument was %T but should have been a string", strAny)
+				}
+
+				u64, err := strconv.ParseUint(str, 10, 64)
+				if err != nil {
+					return types.Bool(false) // A string that's not parseable as a uint is not a valid project number
+				}
+				return types.Bool(u64 >= minProjectNum)
```

Comment:
Would it be better/easier to regex match on 0-9 or \d?

Reply:
Yeah, that would work, but it would mean that the implementation for `gcp_matches_project_number(string)` would be substantially different than `gcp_matches_project_number(int)`. It seems cleaner to have the two overloaded functions be as similar as possible. And I don't think we'd want to use regex for the int implementation; using a regex to check the magnitude of an integer seems gross.

If we were only providing one version of this function that accepted string, then I'd probably do it that way.